src/test/java/net/sf/javaanpr/recognizer/PatternComparerTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2013 JavaANPR contributors
+ * Copyright 2006 Ondrej Martinsky
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.osedu.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package net.sf.javaanpr.recognizer;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class PatternComparerTest {
+
+    @Test
+    public void testCompareAscendingPatternsEqualReturnsZero() {
+        RecognizedPattern recognizedPattern1 = new RecognizedPattern('A', 1.0f);
+        RecognizedPattern recognizedPattern2 = new RecognizedPattern('B', 1.0f);
+        assertEquals(new PatternComparer(0).compare(recognizedPattern1, recognizedPattern2), 0);
+    }
+
+    @Test
+    public void testCompareAscendingFirstPatternSmallerReturnsMinusOne() {
+        RecognizedPattern recognizedPattern1 = new RecognizedPattern('A', 1.0f);
+        RecognizedPattern recognizedPattern2 = new RecognizedPattern('A', 2.0f);
+        assertEquals(new PatternComparer(0).compare(recognizedPattern1, recognizedPattern2), -1);
```

Comment:
Optional: A more robust way of testing a comparator wold be to test for equality with `0`, `< 0` and `> 0`. See the definition of [Comparable](https://docs.oracle.com/javase/6/docs/api/java/lang/Comparable.html) for more info.